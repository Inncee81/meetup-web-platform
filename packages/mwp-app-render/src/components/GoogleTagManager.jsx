// @flow

import React from 'react';

// Production and dev keys
const GTM_KEY = process.env.NODE_ENV === 'production' ? 'GTM-T2LNGD' : 'GTM-W9W847';

/*
 * @description Gets google tag manager JS snippet
 * @see {@link https://developers.google.com/tag-manager/quickstart}
*/
export const getGoogleTagManagerSnippet = (): string => (
	`(function(w,d,s,l,i){
		w[l]=w[l]||[];
		w[l].push({'gtm.start':new Date().getTime(),event:'gtm.js'});
		var f=d.getElementsByTagName(s)[0],j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';
		j.async=true;j.src='https://www.googletagmanager.com/gtm.js?id='+i+dl;
		f.parentNode.insertBefore(j,f);
	})(window,document,'script','dataLayer','${GTM_KEY}');`
);

/*
 * @description Gets google tag manager noscript tag
 * @see {@link https://developers.google.com/tag-manager/quickstart}
*/
export const GoogleTagManagerNoscript = () => {
	const iframe = `
		<iframe
			src="https://www.googletagmanager.com/ns.html?id=${GTM_KEY}"
			height="0" width="0" style="display:none;visibility:hidden">
		</iframe>`;
	return <noscript dangerouslySetInnerHTML={{ __html: iframe }}></noscript>
};
