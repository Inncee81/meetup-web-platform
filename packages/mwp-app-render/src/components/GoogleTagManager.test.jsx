import React from 'react';
import { shallow } from 'enzyme';
import GoogleTagManager from './GoogleTagManager';

const renderComponent = (props = {}) => {
	return shallow(<GoogleTagManager {...props} />);
};

describe('google tag manager', () => {
	it('returns null if rtf is false', () => {
		expect(renderComponent({rtf: false}).equals(null)).toBe(true);
	});
	it('returns null if `tag` prop is undefined or not valid', () => {
		expect(renderComponent({rtf: false}).equals(null)).toBe(true);
		expect(renderComponent({rtf: false, tag: 'div'}).equals(null)).toBe(true);
	});
	describe('script tag', () => {
		const props = {
			rtf: true,
			tag: 'script',
		};
		it('matches snap prod environment', () => {
			expect(renderComponent({...props, isProd: true})).toMatchSnapshot();
		});
		it('matches snap for dev environment', () => {
			expect(renderComponent({...props, isProd: false})).toMatchSnapshot();
		});
	});
	describe('noscript tag', () => {
		const props = {
			rtf: true,
			tag: 'noscript',
		}; 
		it('matches snap prod environment', () => {
			expect(renderComponent({...props, isProd: true})).toMatchSnapshot();
		});
		it('matches snap for dev environment', () => {
			expect(renderComponent({...props, isProd: false})).toMatchSnapshot();
		});
	});
});
