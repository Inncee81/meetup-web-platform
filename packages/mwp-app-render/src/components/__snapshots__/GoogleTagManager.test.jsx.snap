// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`google tag manager noscript tag matches snap for dev environment 1`] = `
<noscript
  dangerouslySetInnerHtml={
    Object {
      "__html": "
    		<iframe
    			src=\\"https://www.googletagmanager.com/ns.html?id=GTM-W9W847\\"
    			height=\\"0\\" width=\\"0\\" style=\\"display:none;visibility:hidden\\">
    		</iframe>",
    }
  }
/>
`;

exports[`google tag manager noscript tag matches snap prod environment 1`] = `
<noscript
  dangerouslySetInnerHtml={
    Object {
      "__html": "
    		<iframe
    			src=\\"https://www.googletagmanager.com/ns.html?id=GTM-T2LNGD\\"
    			height=\\"0\\" width=\\"0\\" style=\\"display:none;visibility:hidden\\">
    		</iframe>",
    }
  }
/>
`;

exports[`google tag manager script tag matches snap for dev environment 1`] = `
<script
  dangerouslySetInnerHtml={
    Object {
      "__html": "
    		(function(w,d,s,l,i){
    			w[l]=w[l]||[];
    			w[l].push({'gtm.start':new Date().getTime(),event:'gtm.js'});
    			var f=d.getElementsByTagName(s)[0],j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';
    			j.async=true;j.src='https://www.googletagmanager.com/gtm.js?id='+i+dl;
    			f.parentNode.insertBefore(j,f);
    		})(window,document,'script','dataLayer','GTM-W9W847');",
    }
  }
/>
`;

exports[`google tag manager script tag matches snap prod environment 1`] = `
<script
  dangerouslySetInnerHtml={
    Object {
      "__html": "
    		(function(w,d,s,l,i){
    			w[l]=w[l]||[];
    			w[l].push({'gtm.start':new Date().getTime(),event:'gtm.js'});
    			var f=d.getElementsByTagName(s)[0],j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';
    			j.async=true;j.src='https://www.googletagmanager.com/gtm.js?id='+i+dl;
    			f.parentNode.insertBefore(j,f);
    		})(window,document,'script','dataLayer','GTM-T2LNGD');",
    }
  }
/>
`;
